server/channels/app/server.go

```diff
@@ -1871,3 +1868,125 @@ func (s *Server) Log() *mlog.Logger {
 	return s.platform.NotificationsLogger()
 }
+
+func (a *App) initElasticsearchChannelIndexCheck() {
+	a.elasticsearchChannelIndexCheckWithRetry()
+
+	a.AddConfigListener(func(oldConfig, newConfig *model.Config) {
+		if a.SearchEngine().ElasticsearchEngine == nil {
+			return
+		}
+
+		oldESConfig := oldConfig.ElasticsearchSettings
+		newESConfig := newConfig.ElasticsearchSettings
+
+		// if indexing is turned on, check.
+		if !*oldESConfig.EnableIndexing && *newESConfig.EnableIndexing {
+			a.elasticsearchChannelIndexCheckWithRetry()
+		} else if *newESConfig.EnableIndexing && (*oldESConfig.Password != *newESConfig.Password || *oldESConfig.Username != *newESConfig.Username || *oldESConfig.ConnectionURL != *newESConfig.ConnectionURL || *oldESConfig.Sniff != *newESConfig.Sniff) {
+			// ES client reconnects if credentials or address changes
+			a.elasticsearchChannelIndexCheckWithRetry()
+		}
+	})
+
+	a.AddLicenseListener(func(oldLicense, newLicense *model.License) {
+		if a.SearchEngine() == nil {
+			return
+		}
+
+		// if a license was added, and it has ES enabled-
+		if oldLicense == nil && newLicense != nil {
+			if a.SearchEngine().ElasticsearchEngine != nil {
+				a.elasticsearchChannelIndexCheckWithRetry()
+			}
+		}
+	})
+}
+
+func (a *App) elasticsearchChannelIndexCheckWithRetry() {
+	go func() {
+		// using progressive retry because ES client may take some time to connect and be ready.
+		_ = utils.ProgressiveRetry(func() error {
+			if !*a.Config().ElasticsearchSettings.EnableIndexing {
+				a.Log().Debug("elasticsearchChannelIndexCheckWithRetry: skipping because elasticsearch indexing is disabled")
+				return nil
+			}
+
+			elastic := a.SearchEngine().ElasticsearchEngine
+			if elastic == nil {
+				a.Log().Debug("elasticsearchChannelIndexCheckWithRetry: skipping because elastic engine is nil")
+				return errors.New("retry")
+			}
+
+			if !elastic.IsActive() {
+				a.Log().Debug("elasticsearchChannelIndexCheckWithRetry: skipping because elastic.IsActive is false")
+				return errors.New("retry")
+			}
+
+			a.elasticsearchChannelIndexCheck()
+			return nil
+		})
+	}()
+}
+
+func (a *App) elasticsearchChannelIndexCheck() {
+	if needNotify := a.elasticChannelsIndexNeedNotifyAdmins(); !needNotify {
+		return
+	}
+
+	// notify all system admins
+	systemBot, appErr := a.GetSystemBot()
+	if appErr != nil {
+		a.Log().Error("elasticsearchChannelIndexCheck: couldn't get system bot", mlog.Err(appErr))
+		return
+	}
+
+	sysAdmins, appErr := a.getAllSystemAdmins()
+	if appErr != nil {
+		a.Log().Error("elasticsearchChannelIndexCheck: error occurred fetching all system admins", mlog.Err(appErr))
+	}
+
+	elasticsearchSettingsSectionLink, err := url.JoinPath(*a.Config().ServiceSettings.SiteURL, "admin_console/environment/elasticsearch")
+	if err != nil {
+		a.Log().Error("elasticsearchChannelIndexCheck: error occurred constructing Elasticsearch system console section path")
+		return
+	}
+
+	// TODO include a link to changelog
```

Comment:
Is this going to be addressed now? If not, should we create a ticket to address this? (and link the ticket here)

Reply:
It will be done before PR is merged.